module-system/sov-state/src/vec.rs

```diff
@@ -0,0 +1,218 @@
+use std::iter::FusedIterator;
+use std::marker::PhantomData;
+
+use thiserror::Error;
+
+use crate::codec::{BorshCodec, PairOfCodecs, StateValueCodec};
+use crate::{Prefix, StateMap, StateValue, Storage, WorkingSet};
+
+type InternalCodec<C> = PairOfCodecs<BorshCodec, C>;
+
+#[derive(Debug, Clone)]
+pub struct StateVec<V, C = BorshCodec>
+where
+    C: StateValueCodec<V>,
+{
+    _phantom: PhantomData<V>,
+    prefix: Prefix,
+    len_value: StateValue<usize>,
+    elems: StateMap<usize, V, InternalCodec<C>>,
```

Comment:
(non-blocking) It seems possible that some use cases will also want control over the key codec. Will it be trivial to switch from a `StateValueCodec` to a `StateCodec` if that turns out to be the case?